numpy/core/src/multiarray/datetime.c

```diff
@@ -1761,6 +1762,9 @@ parse_datetime_unit_from_string(char *str, Py_ssize_t len, char *metastr)
                 return NPY_FR_as;
         }
     }
+    else if (allow_generic && len == 7 && !strncmp(str, "generic", 7)) {
```

Comment:
> Not sure why that would be a problem

I'm not sure either, but I'm not germane to the design decisions that were made, which is why I was cautious with this.
The argument of not changing accepted input in a bugfix release sounds more serious to me, what do you think?